src/Gren/Kernel/FileSystem.js

```diff
@@ -217,3 +218,32 @@ var _FileSystem_listDirectoryContent = function (path) {
     });
   });
 };
+
+// List of dir contents as DirEntry values holding filename string
+var _FileSystem_listDirectory = function (path) {
+  return __Scheduler_binding(function (callback) {
+    fs.readdir(path, { withFileTypes: true }, function (err, content) {
+      if (err != null) {
+        callback(__Scheduler_fail(_FileSystem_constructAccessError(err)));
+      } else {
+        callback(__Scheduler_succeed(content.map(toGrenDirEntry)));
+      }
+    });
+  });
+};
+
+var toGrenDirEntry = function (dirEnt) {
```

Comment:
Would you mind appending `_FileSystem_` here? Kernel code isn't namespaced, so it could potentially clash with another similarly named kernel code function (however unlikely).

Reply:
Definitely.

Does this mean there's no clean way to keep a function "private" to the kernel?

Reply:
Yes.